documentation/latest/dev/installation.md

```diff
@@ -0,0 +1,214 @@
+  install-router   Install openshift router
+  controller-gen   Download controller-gen locally if necessary.
+  kustomize        Download kustomize locally if necessary.
+  ginkgo           Download ginkgo locally if necessary.
+  client-gen       Download client-gen locally if necessary.
+  lister-gen       Download lister-gen locally if necessary.
+  informer-gen     Download client-gen locally if necessary.
+  k8s-client-gen   Generate typed client for flotta project.
+  validate-swagger  Validate swagger
+$ ->
+```
+
+
+## Operator installation
+
+For running operator, you need to install the CRD first, the best way to develop
+is in a small Kubernetes cluster, like Kind or Minikube.
+
+To install Operator you need to think that there are two main part, the CRD
+installation and the operator itself.
+
+To install CRD, you need to run the folowing:
+
+```
+$ -> make install
+/home/eloy/dev/upstream/project-flotta/k4e-operator/bin/kustomize build config/crd | kubectl apply -f -
+customresourcedefinition.apiextensions.k8s.io/edgeconfigs.management.project-flotta.io configured
+customresourcedefinition.apiextensions.k8s.io/edgedevices.management.project-flotta.io configured
+customresourcedefinition.apiextensions.k8s.io/edgedevicesets.management.project-flotta.io configured
+customresourcedefinition.apiextensions.k8s.io/edgedevicesignedrequest.management.project-flotta.io configured
+customresourcedefinition.apiextensions.k8s.io/edgeworkloads.management.project-flotta.io configured
+customresourcedefinition.apiextensions.k8s.io/playbookexecutions.management.project-flotta.io configured
+```
+
+To run the operator, some dependencies need to be executed first, like
+openshift-router and cert-manager:
+
+```shell
+make install-router
+make install-cert-manager
+```
+
+To run operator locally, you can use the following code:
+
+```shell
+make run
+```
+
+For registering devices, it's important to download certs and set correct
+persmission:
+
+```shell
+$ -> sudo rm /tmp/*.pem
+$ -> make get-certs
+kubectl get secret -n flotta flotta-ca  -o json | jq '.data."ca.crt"| @base64d' -r >/tmp/ca.pem
+kubectl -n flotta get secret reg-client-ca-jnpmz68qfr -o json | jq -r '.data."client.crt"| @base64d' > /tmp/cert.pem
+kubectl -n flotta get secret reg-client-ca-jnpmz68qfr -o json | jq -r '.data."client.key"| @base64d' > /tmp/key.pem
+$ -> sudo chown root:root /tmp/*.pem
+```
+
+## Device installation
+
+On device, we have two different things to install, yggdrasil and device worker.
+There are two ways to run this piece:
+
+> ⚠️ **Golang 1.16**: By default, device needs to compile in Golang 1.16
+> mainly because this software needs to packaged as RPM, so the version for
+> RHEL8/Fedora 35 is golang 1.16
+
+### Single way with Yggdrasil:
+
+This way, means that Yggdrasil already knows where the device-worker is located,
+and it's defined on workers.toml.
+
+So for this,on device-worker repo the following command should be run:
```

Comment:
I would mention that now we run it from worker repo. Should we add make help call for the device?

Reply:
not sure if I follow you here.